cli/admin/src/main/java/io/pravega/cli/admin/utils/ZKHelper.java

```diff
@@ -112,6 +117,22 @@ private ZKHelper(String zkURL, String clusterName) throws ZKConnectionFailedExce
 
     }
 
+    /**
+     * Get the metadata information of reader for a given path.
+     * @param readerPath path of reader
+     * @throws Exception if unable to get the znode for readerpath.
+     * @return detailed metadata information of reader
+     */
+    public String getMetaDataForReader(String readerPath) throws Exception {
```

Comment:
Is it possible to be specific about the Exception thrown here and how is it handled?

Reply:
forPath() method throws Exception thats why proceed it further. It is handled [here](https://github.com/Bhupender-Y/pravega/blob/issue-6311-admin-cli-command-to-view-reader-info/cli/admin/src/main/java/io/pravega/cli/admin/controller/metadata/ControllerMetadataViewReaderInfoCommand.java#L41)

Reply:
what gets show in cli in case of exception?

Reply:
can you check the behavior in case of exception via the test case?

Reply:
When an exception occurs, **Exception accessing to reader metadata** message shows in cli. It is handled [here](https://github.com/Bhupender-Y/pravega/blob/issue-6311-admin-cli-command-to-view-reader-info/cli/admin/src/main/java/io/pravega/cli/admin/controller/metadata/ControllerMetadataViewReaderInfoCommand.java#L41).  To verify exception, **testControllerMetadataViewReaderInfoCommandWithException** is written in [AbstractControllerMetadataCommandsTest ](https://github.com/Bhupender-Y/pravega/blob/issue-6311-admin-cli-command-to-view-reader-info/cli/admin/src/test/java/io/pravega/cli/admin/controller/AbstractControllerMetadataCommandsTest.java#L331).